packages/notarization/manager.go

```diff
@@ -469,12 +486,53 @@ func (m *Manager) outputsToOutputIDs(outputs devnetvm.Outputs) (createdIDs utxo.
 	return
 }
 
+// CheckIfEpochChanged check if next epoch started and trigger the event.
+func (m *Manager) CheckIfEpochChanged(issuingTime time.Time) {
+	currentTime := m.tangle.TimeManager.ATT()
+	if issuingTime.After(currentTime) {
+		currentTime = issuingTime
+	}
+	ei := epoch.IndexFromTime(currentTime)
+	currentEpochIndex, err := m.epochCommitmentFactory.storage.CurrentEpochIndex()
```

Comment:
Isn't this equivalent to checking the last epoch we committed to? So LastCommittedEpoch?

Reply:
no, what if the node was not sending messages, then the last committed epoch stayed unchanged, as we update committments when we call `GetLatestEC()`. currentEpochIndex is advancing even if we do not create committments